internal/dataplane/kongstate/kongstate.go

```diff
@@ -225,6 +230,18 @@ func (ks *KongState) FillUpstreamOverrides(s store.Storer, failuresCollector *fa
 			failuresCollector.PushResourceFailure(err.Error(), lo.Map(servicesGroup, servicesAsObjects)...)
 		} else {
 			if kongUpstreamPolicy != nil {
+				if kongIngress != nil {
+					for _, svc := range servicesGroup {
+						logger.Error(nil,
+							fmt.Sprintf("Service uses both %s and %s annotations, should use only %s annotation. Settings "+
+								"from %s will take precedence",
+								annotations.AnnotationPrefix+annotations.ConfigurationKey,
+								annotations.AnnotationPrefix+kongv1beta1.KongUpstreamPolicyAnnotationKey,
+								annotations.AnnotationPrefix+kongv1beta1.KongUpstreamPolicyAnnotationKey,
+								annotations.AnnotationPrefix+kongv1beta1.KongUpstreamPolicyAnnotationKey),
+							"namespace", svc.Namespace, "name", svc.Name)
+					}
+				}
```

Comment:
No need for that @rainest as we're going to deprecate KongIngress entirely anyway: https://github.com/Kong/kubernetes-ingress-controller/pull/5015

Reply:
Are we deprecating it or removing it? I thought we were going to keep it for a version or two since we hadn't introduced KongUpstreamPolicy before 3.0.

Removing it immediately would pose an upgrade challenge: you can't upgrade to 3.0 while keeping configuration managed through existing KongIngresses intact, and you can't start using KongUpstreamPolicy until you've already upgraded to 3.0.

You could technically install the CRD and create replacement resources/annotations prior to upgrading, but it'd be risky since you can't test them before committing to the upgrade. IMO we should allow a window where you can use both the old and new resources.

If we do end up removing KongIngress entirely in 3.0, I suppose we can toss a revert for this commit into #5015, even if it's a bit silly to have then added it in the first place.

Reply:
Other changes were requested anyway, so removed my commits here and moved them into https://github.com/Kong/kubernetes-ingress-controller/pull/5022

Reply:
We discussed this today with @mflendrich and the decision was to:
1) stop allowing users to set `proxy` and `route` fields 
2) deprecate `upstream` (still allow, but return a warning from the admission webhook and do not honor the configuration)

I see your point about allowing smoother migration and I'd be okay with doing it this way. @mflendrich what do you think?

That change the above points to:
1) stop allowing users to set `proxy` and `route` fields 
2) deprecate `upstream` (return a warning from the admission webhook and still handle configuration)
3) create an issue to track disallowing `upstream` field to be set in KIC 3.1?

Reply:
Turned out it was a miscommunication. We're going to go https://github.com/Kong/kubernetes-ingress-controller/pull/5022 way.